src/spatialdata/_io/accessor.py

```diff
@@ -0,0 +1,136 @@
+
+    @property
+    def data(self) -> Any:
+        """Access the raw internal attrs dict."""
+        return self._obj._attrs
+
+
+@register_dataframe_accessor("attrs")
+class DfAttrsAccessor(_AttrsBase):
+    """Dict-like .attrs accessor for Dask DataFrames."""
+
+    pass
+
+
+@register_series_accessor("attrs")
+class SeriesAttrsAccessor(_AttrsBase):
+    """Dict-like .attrs accessor for Dask Series."""
+
+    pass
+
+
+def wrap_with_attrs(method: Callable[..., Any]) -> Callable[..., Any]:
+    """Wrap a Dask DataFrame method to preserve _attrs.
+
+    Copies _attrs from self before calling method, then assigns to result.
+    Safe for lazy operations like set_index, assign, map_partitions.
+    """
+
+    def wrapper(self: Any, *args: Any, **kwargs: Any) -> Any:
+        if not isinstance(self.attrs, DfAttrsAccessor | SeriesAttrsAccessor):
+            raise RuntimeError(
+                "Invalid .attrs: expected an accessor (DfAttrsAccessor or SeriesAttrsAccessor), "
+                f"got {type(self.attrs).__name__}. A common cause is assigning a dict, e.g. "
+                "my_dd_object.attrs = {...}. Do not assign to 'attrs'; use "
+                "my_dd_object.attrs.update(...) instead."
+            )
+
+        old_attrs = self.attrs.copy()
+        result = method(self, *args, **kwargs)
+        result.attrs.update(old_attrs)
+        return result
+
+    return wrapper
+
+
+def wrap_indexer_with_attrs(indexer_name: Literal["loc", "iloc"]) -> None:
+    """Patch dd.DataFrame.loc or iloc to preserve _attrs.
+
+    Reason for having this separate from methods is because both loc and iloc are a property that return an indexer.
+    Therefore, they have to be wrapped differently from methods in order to preserve attrs.
+    """
+    original_property = getattr(dd.DataFrame, indexer_name)  # this is a property
+
+    def indexer_with_attrs(self: dd.DataFrame) -> Any:
+        df = self
+        loc = original_property.fget(df)
+
+        class IndexerWrapper:
+            def __init__(self, parent_loc: Any, parent_df: dd.DataFrame) -> None:
+                self._parent_loc = parent_loc
+                self._parent_df = parent_df
+
+            def __getitem__(self, key: str) -> Any:
+                result = self._parent_loc[key]
+                if hasattr(self._parent_df, "attrs"):
+                    result._attrs = self._parent_df.attrs.copy()
+                return result
+
+            def __setitem__(self, key: str, value: Any) -> dd.DataFrame:
+                # preserve attrs even if user assigns via .loc
+                self._parent_loc[key] = value
+                return self._parent_df
+
+            def __repr__(self) -> str:
+                return repr(self._parent_loc)
+
+        return IndexerWrapper(loc, df)
+
+    setattr(dd.DataFrame, indexer_name, property(indexer_with_attrs))
+
+
+methods_to_wrap = [
+    "set_index",
+    "compute",
+    "drop",
+    "__getitem__",
+    "copy",
+    "cat",
```

Comment:
cat is never used to wrap an accessor because hasattr(dd.DataFrame, 'cat') is False.

Reply:
We can proceed as follows:
1. we do not wrap `cat` (everything seems to work)
2. if we see we need to wrap `cat`, let's modify `wrap_method_with_attrs` to access methods from `dd.Series` (now it uses `getattrs` and `setattrs` with `dd.DataFrame`).

Reply:
Actually 1 is the way to go: `pandas.Series` do not have `.attrs` in `.cat`.